src/shared/qrcode-util.c

```diff
@@ -116,4 +185,46 @@ int print_qrcode(FILE *out, const char *header, const char *string) {
         sym_QRcode_free(qr);
         return 0;
 }
+
+int print_positioned_qrcode(FILE *out, const char *header, const char *string, unsigned row, unsigned column, unsigned tty_width, unsigned tty_height) {
+        QRcode* qr;
+        int r, fd;
+        unsigned qr_code_width, qr_code_height;
+
+        fd = fileno(out);
+        if (fd < 0)
+                return log_warning_errno(fd, "unable to get file descriptor from the file stream: %m");
+
+        /* If this is not an UTF-8 system or ANSI colors aren't supported/disabled don't print any QR
+         * codes */
+        if (!is_locale_utf8() || !colors_enabled())
+                return -EOPNOTSUPP;
+
+        r = dlopen_qrencode();
+        if (r < 0)
+                return r;
+
+        qr = sym_QRcode_encodeString(string, 0, QR_ECLEVEL_L, QR_MODE_8, 1);
+        if (!qr)
+                return -ENOMEM;
+
+        if (header) {
+                set_terminal_cursor_position(fd, row - 1, column);
+                fprintf(out, "\n%s:\n\n", header);
+        }
+
+        qr_code_width = qr_code_height = qr->width + 8;
+        if (column + qr_code_width > tty_width)
+                column = tty_width - qr_code_width;
+
+        if (row + qr_code_height > tty_height)
+                row = tty_height - qr_code_height;
+
+        write_positioned_qrcode(out, qr, row, column);
```

Comment:
This is what I've used to fix the tty overflow issue.
I tried it with 90%/90%, the QR code survived.

Is it good to go sir.